src/Mod/Sketcher/Gui/DrawSketchDefaultHandler.h

```diff
@@ -0,0 +1,1127 @@
+    {}
+
+    /** @name public DrawSketchHandler interface
+     * NOTE: Not intended to be specialised. It calls some functions intended to be
+     * overridden/specialised instead.
+     */
+    //@{
+    void mouseMove(Base::Vector2d onSketchPos) override
+    {
+        updateDataAndDrawToPosition(onSketchPos);
+    }
+
+    bool pressButton(Base::Vector2d onSketchPos) override
+    {
+
+        onButtonPressed(onSketchPos);
+        return true;
+    }
+
+    bool releaseButton(Base::Vector2d onSketchPos) override
+    {
+        Q_UNUSED(onSketchPos);
+        finish();
+        return true;
+    }
+
+    void registerPressedKey(bool pressed, int key) override
+    {
+        if (key == SoKeyboardEvent::M && pressed && !this->isLastState()) {
+            this->iterateToNextConstructionMethod();
+        }
+    }
+    //@}
+
+
+protected:
+    using SelectMode = SelectModeT;
+    using ModeStateMachine = StateMachine<SelectModeT>;
+    using ConstructionMethod = ConstructionMethodT;
+    using ConstructionMachine = ConstructionMethodMachine<ConstructionMethodT>;
+
+    /** @name functions NOT intended for specialisation or to be hidden
+        These functions define a predefined structure and are extendable using NVI.
+        1. Call them from your handle
+        2. Do not hide them or otherwise redefine them UNLESS YOU HAVE A REALLY GOOD REASON TO
+        3. EXTEND their functionality, if needed, using the NVI interface (or if you do not need to
+       derive, by specialising these functions).*/
+    //@{
+    /** @brief This function finalises the creation operation. It works only if the state machine is
+     * in state End.
+     *
+     * @details
+     * The functionality need to be provided by extending these virtual private functions:
+     * 1. executeCommands() : Must be provided with the Commands to create the geometry
+     * 2. generateAutoConstraints() : When using AutoConstraints vector, this function populates the
+     * AutoConstraints vector, ensuring no redundant autoconstraints
+     * 2. beforeCreateAutoConstraints() : Enables derived classes to define specific actions before
+     * executeCommands and createAutoConstraints (optional).
+     * 3. createAutoConstraints() : Must be provided with the commands to create autoconstraints
+     *
+     * It recomputes if not solves and handles continuous mode automatically
+     */
+    void finish()
+    {
+        if (this->isState(SelectMode::End)) {
+            unsetCursor();
+            resetPositionText();
+
+            try {
+                executeCommands();
+
+                if (sugConstraints.size() > 0) {
+                    generateAutoConstraints();
+
+                    beforeCreateAutoConstraints();
+
+                    createAutoConstraints();
+                }
+
+                tryAutoRecomputeIfNotSolve(
+                    static_cast<Sketcher::SketchObject*>(sketchgui->getObject()));
+            }

```

Comment:
I will handle these three in a separate PR. I am starting to use qtcreator and it seems we have quite some of this issues, which were not detected by kdevelop.